### PR TITLE
fix (index.d.ts): 'of' missing 'route' with unit tests

### PR DESCRIPTION
**Pull request by Maledong**
1) 'of' has missed the method of 'route', and this should be a method
of egg-socket itself. So add it (See: https://github.com/eggjs/egg-socket.io/pull/37#issuecomment-384183665).

2) Add a whole missing tests including a full example of namespaced,
packagemiddlewares, controllmiddlewares...ect to run with ts.

### CODE REVIEW

File: lib/loader.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ module.exports = app => {
     directory: dirs,
     target: app.io.middleware,
     inject: app,
+    typescript: true,
```
Comment (whxaxes) [inline lib/loader.js]: This property has deprecated

Comment (whxaxes) [reply lib/loader.js]: Loader load ts file by env( see https://github.com/eggjs/egg-core/blob/4d4113cfd27d1e8ce4ce65d2d19b0035b5291dcc/lib/loader/file_loader.js#L125 ), this env will be set if you are using `egg-bin dev --ts`

Comment (ngot) [reply lib/loader.js]: `typescript: true` is not necessary here.

Comment (Maledong) [reply lib/loader.js]: Well……Just for a confirm：The problem 'https://github.com/eggjs/egg/issues/2398#issuecomment-386264963' is really fixed? If your answer is YES, and I'll remove this :)

Comment (whxaxes) [reply lib/loader.js]: @Maledong Yes

Comment (Maledong) [reply lib/loader.js]: Thanks! I'll remove and resubmit:)

**Comment by atian25** [comment]
@whxaxes ready to merge ?

**Comment by ngot** [comment]
LGTM

**Comment by ngot** [comment]
+ egg-socket.io@4.1.2

**Pull request merged**